server/plugins/messageStorage/sqlite.ts

```diff
@@ -243,31 +227,59 @@ class SqliteMessageStorage implements ISqliteMessageStorage {
 		query.offset = parseInt(query.offset as string, 10) || 0;
 		params.push(String(query.offset));
 
+		const rows = await this.serialize_fetchall(select, ...params);
+		const response: SearchResponse = {
+			searchTerm: query.searchTerm,
+			target: query.channelName,
+			networkUuid: query.networkUuid,
+			offset: query.offset as number,
+			results: parseSearchRowsToMessages(query.offset as number, rows).reverse(),
+		};
+
+		return response;
+	}
+
+	canProvideMessages() {
+		return this.isEnabled;
+	}
+
+	private run(stmt: string, ...params: any[]) {
+		this.serialize_run(stmt, params).catch((err) =>
+			log.error(`failed to run ${stmt}`, String(err))
+		);
+	}
+
+	private serialize_run(stmt: string, params: any[]): Promise<void> {
 		return new Promise((resolve, reject) => {
-			this.database.all(select, params, (err, rows) => {
-				if (err) {
+			this.database.serialize(() => {
+				this.database.run(stmt, params, (err) => {
+					if (!err) {
```

Comment:
I think this could be a little simpler.

```typescript
if (err) {
    reject(err);
} else {
    resolve();
}
```

Reply:
ack, patched in the patchset that will succeed this one